src/pages/workspaces/workspace/Dashboard/Dashboard.ts

```diff
@@ -0,0 +1,511 @@
+    backgroundColor: colors.dark(0.15),
+    borderRadius: 10,
+    overflow: 'hidden',
+    wordWrap: 'break-word',
+  },
+};
+
+const DashboardAuthContainer = (props: WorkspaceDashboardProps): ReactNode => {
+  const { namespace, name } = props;
+  const { signInStatus } = useStore(authStore);
+  const [featuredWorkspaces, setFeaturedWorkspaces] = useState<{ name: string; namespace: string }[]>();
+
+  const isAuthInitialized = signInStatus !== 'uninitialized';
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setFeaturedWorkspaces(await Ajax().FirecloudBucket.getFeaturedWorkspaces());
+    };
+    if (signInStatus === 'signedOut') {
+      fetchData();
+    }
+  }, [signInStatus]);
+
+  const isFeaturedWorkspace = () => _.some((ws) => ws.namespace === namespace && ws.name === name, featuredWorkspaces);
+
+  return cond(
+    [
+      !isAuthInitialized || (signInStatus === 'signedOut' && featuredWorkspaces === undefined),
+      () => centeredSpinner({ style: { position: 'fixed' } }),
+    ],
+    [signInStatus === 'signedOut' && isFeaturedWorkspace(), () => h(DashboardPublic, props)],
+    [signInStatus === 'signedOut', () => h(SignIn)],
+    () => h(WorkspaceDashboard, props)
+  );
+};
+
+interface WorkspaceDashboardProps {
+  namespace: string;
+  name: string;
+}
+
+interface WorkspaceDashboardComponentProps extends WorkspaceDashboardProps {
+  refreshWorkspace: () => void;
+  storageDetails: StorageDetails;
+  workspace: Workspace;
+}
+
+const WorkspaceDashboardComponent = (
+  props: WorkspaceDashboardComponentProps,
+  ref: ForwardedRef<unknown>
+): ReactNode => {
+  const {
+    namespace,
+    name,
+    refreshWorkspace,
+    storageDetails,
+    workspace,
+    workspace: {
+      accessLevel,
+      owners = [],
+      workspace: { authorizationDomain, attributes = { description: '' } },
+    },
+  } = props;
+
+  const description = attributes.description;
+
+  // State
+  const [storageCost, setStorageCost] = useState<{ isSuccess: boolean; estimate: string; lastUpdated?: string }>();
+  const [bucketSize, setBucketSize] = useState<{ isSuccess: boolean; usage: string; lastUpdated?: string }>();
+  const [editDescription, setEditDescription] = useState<string>();
+  const [saving, setSaving] = useState<boolean>(false);
+  const [busy, setBusy] = useState<boolean>(false);
+  const [consentStatus, setConsentStatus] = useState<string>();
+  const [tagsList, setTagsList] = useState<string[]>();
+  const [acl, setAcl] = useState<WorkspaceAcl>();
+
+  const persistenceId = `workspaces/${namespace}/${name}/dashboard`;
+
+  const signal = useCancellation();
+
+  const refresh = () => {
+    loadConsent();
+    loadWsTags();
+
+    // If the current user is the only owner of the workspace, load the ACL to check if the workspace is shared.
+    if (isOwner(accessLevel) && _.size(owners) === 1) {
+      loadAcl();
+    }
+    updateGoogleBucketDetails(workspace);
+  };
+
+  const loadStorageCost = useMemo(
```

Comment:
Many of these `load...` functions have some extra logic that makes them particularly good candidates for custom hooks.